src/main/java/net/coderbot/iris/pipeline/HandRenderer.java

```diff
@@ -0,0 +1,130 @@
+import net.minecraft.client.Camera;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.GameRenderer;
+import net.minecraft.client.renderer.ItemBlockRenderTypes;
+import net.minecraft.client.renderer.RenderBuffers;
+import net.minecraft.client.renderer.RenderType;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.entity.EquipmentSlot;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.level.GameType;
+
+public class HandRenderer {
+	public static final HandRenderer INSTANCE = new HandRenderer();
+
+	private static boolean ACTIVE;
+	public static boolean renderingSolid;
+
+	private void setupGlState(GameRenderer gameRenderer, PoseStack poseStack, float tickDelta, Camera camera) {
+        final PoseStack.Pose pose = poseStack.last();
+
+		// We have a inject in getProjectionMatrix to scale the matrix so the hand doesn't clip through blocks.
+		gameRenderer.resetProjectionMatrix(gameRenderer.getProjectionMatrix(camera, tickDelta, false));
+
+        pose.pose().setIdentity();
+        pose.normal().setIdentity();
+
+		if(Minecraft.getInstance().options.bobView) {
+			((GameRendererAccessor)gameRenderer).invokeBobView(poseStack, tickDelta);
+		}
+	}
+
+	private boolean canRender(Camera camera, GameRenderer gameRenderer) {
+		return !(camera.isDetached() 
+			|| !(camera.getEntity() instanceof Player) 
+				|| ((GameRendererAccessor)gameRenderer).getPanoramicMode() 
+					|| Minecraft.getInstance().options.hideGui 
+						|| (camera.getEntity() instanceof LivingEntity && ((LivingEntity)camera.getEntity()).isSleeping()) 
+							|| Minecraft.getInstance().gameMode.getPlayerMode() == GameType.SPECTATOR);
+	}
+
+	public boolean isHandTranslucent(InteractionHand hand) {
+		if (Minecraft.getInstance().player.getItemBySlot(hand == InteractionHand.OFF_HAND ? EquipmentSlot.OFFHAND : EquipmentSlot.MAINHAND).getItem() instanceof BlockItem) {
+			return ItemBlockRenderTypes.getChunkRenderType(((BlockItem) Minecraft.getInstance().player.getItemBySlot(hand == InteractionHand.OFF_HAND ? EquipmentSlot.OFFHAND : EquipmentSlot.MAINHAND).getItem()).getBlock().defaultBlockState()) == RenderType.translucent();
+		}
+
+		return false;
+	}
+
+	public boolean isAnyHandTranslucent() {
+		return isHandTranslucent(InteractionHand.MAIN_HAND) || isHandTranslucent(InteractionHand.OFF_HAND);
+	}
+
+	public void renderSolid(RenderBuffers renderBuffers, PoseStack poseStack, float tickDelta, Camera camera, GameRenderer gameRenderer, WorldRenderingPipeline pipeline) {
+		if(!canRender(camera, gameRenderer)) {
+			return;
+		}
+
+		ACTIVE = true;
+
+		poseStack.pushPose();
+
+		Minecraft.getInstance().getProfiler().push("iris_hand");
+
+		setupGlState(gameRenderer, poseStack, tickDelta, camera);
+
+		pipeline.pushProgram(GbufferProgram.HAND);
+
+		RenderSystem.disableBlend();
```

Comment:
This generally has no effect since render layers always override blend modes. You can probably just remove it.